nifi-commons/nifi-flow-encryptor/src/main/java/org/apache/nifi/flow/encryptor/JsonFlowEncryptor.java

```diff
@@ -33,9 +34,13 @@ public class JsonFlowEncryptor extends AbstractFlowEncryptor {
     @Override
     public void processFlow(final InputStream inputStream, final OutputStream outputStream,
                             final PropertyEncryptor inputEncryptor, final PropertyEncryptor outputEncryptor) {
+
+        final BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+        bufferedInputStream.mark(1);
```

Comment:
This addition does not appear necessary. The only reason for it in the standard class was to support content type detection.